kore/src/Kore/Step/Axiom/EvaluationStrategy.hs

```diff
@@ -86,7 +89,8 @@ definitionEvaluation
     -> BuiltinAndAxiomSimplifier
 definitionEvaluation rules =
     BuiltinAndAxiomSimplifier
-        (evaluateWithDefinitionAxioms rules)
+        (\_ _ _ term _ ->
```

Comment:
I think that I would use a different heuristic: whenever I have to add a parameter of type `Predicate` and the function uses `topTODO`, I would immediately use it instead. So in this case, I would give a name to that new parameter and replace `topTODO` with it.